### DISCUSSION TITLE
The field ctx is not a part of ValidationError schema

### DISCUSSION THREAD

**Discussion by Hultner** [fastapi/fastapi#8948] [Questions]
ValidationErrors actually include a ctx parameter, this is great and useful but it's unfortunately not documented by the OpenAPI-schema which causes problems with tools relying on schema correctness as well as confusion for API consumers. I'd categorise this as bug considering it's most likely unintentional undocumented behaviour.

### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI
from pydantic import Body

app = FastAPI()


@app.post("/")
def post_undocumented(
    data_param: str = Body(..., min_length=16)
):
    # Returned value is irrelevant for example
    return {"Hello": "World"}
```

### Description

* Open the browser and call post to the endpoint `/`, using `{ "data_param": "a" }`.
* It returns a JSON with validation error containing the field `ctx`.
* But I expected it to return a object matching openapi.json's represenation of validation error.
```json
{
  "title": "ValidationError",
  "required": [
    "loc",
    "msg",
    "type"
  ],
  "type": "object",
  "properties": {
    "loc": {
      "title": "Location",
      "type": "array",
      "items": {
        "type": "string"
      }
    },
    "msg": {
      "title": "Message",
      "type": "string"
    },
    "type": {
      "title": "Error Type",
      "type": "string"
    }
  }
}
```

### Environment

* OS: [e.g. Linux / Windows / macOS]: N/A, but Linux
* FastAPI Version [e.g. 0.3.0]: 0.54.2
* Python version: 3.8

### Additional context

Don't think any additional information is needed

**Comment by Mause** [comment]
Looks like it needs to be added here: https://github.com/tiangolo/fastapi/blob/master/fastapi/openapi/utils.py#L33

**Comment by Hultner** [comment]
> Looks like it needs to be added here: https://github.com/tiangolo/fastapi/blob/master/fastapi/openapi/utils.py#L33

Yep, not sure what type it should be though, looks like it can differ depending on the error, so I'm not quite sure what the ctx-type should look like, or if should be some sort of `AnyValue`/`anyOf`

**Comment by Hultner** [comment]
I just saw that I somehow labelled this as a question which it probably shouldn't be, but I can't fix it now.

**Comment by marcelocarmona** [comment]
Is there an universal solution available for this issue in FastAPI version 0.95.1?


<img width="902" alt="Screenshot 2023-10-04 at 10 32 27 AM" src="https://github.com/tiangolo/fastapi/assets/4614400/d119e19d-95dd-4ccc-9cb1-40052f1bc35c">

**Comment by YuriiMotov** [comment]
In current version (0.115.14) it's changed a bit.
Now if you pass `{ "data_param": "a" }`, then `ctx` will not be included in the response.

```json
{
  "detail": [
    {
      "loc": [
        "body"
      ],
      "msg": "str type expected",
      "type": "type_error.str"
    }
  ]
}
```

But the problem is still reproducible with the request containing wrong json: `{ "data_param": "a"`:
```json
{
  "detail": [
    {
      "type": "json_invalid",
      "loc": [
        "body",
        20
      ],
      "msg": "JSON decode error",
      "input": {},
      "ctx": {
        "error": "Expecting ',' delimiter"
      }
    }
  ]
}
```

Here, `input` and `ctx` fields violate the schema, which now is looking like:

```json
"ValidationError": {
        "properties": {
          "loc": {
            "items": {
              "anyOf": [
                {
                  "type": "string"
                },
                {
                  "type": "integer"
                }
              ]
            },
            "type": "array",
            "title": "Location"
          },
          "msg": {
            "type": "string",
            "title": "Message"
          },
          "type": {
            "type": "string",
            "title": "Error Type"
          }
        },
        "type": "object",
        "required": [
          "loc",
          "msg",
          "type"
        ],
        "title": "ValidationError"
      }
    }
```

**Comment by YuriiMotov** [comment]
There is an issue related to this: https://github.com/fastapi/fastapi/issues/10787
Let's track it there
